### PR TITLE
Ts 204 modify user to to support list

### PR DESCRIPTION
**Pull request by rhiGriff**
**What is the context of this PR?**

Link to Trello card
https://trello.com/c/iZMW18em
Describe what you have changed and why, link to other PRs or Issues as appropriate.
Changed the msg_to to accept a list of strings.

**How to review**

Any notes for the reviewer  (include screenshots if appropriate).

### CODE REVIEW

File: app/common/utilities.py [hunk 1]
```diff
@@ -95,11 +95,13 @@ def paginated_list_to_json(paginated_list, page, limit, host_url, user, string_q
 
 def generate_etag(msg_to, msg_id, subject, body):
     """Function used to create an ETag"""
-    if None in msg_to:
+    if msg_to is None:
         msg_to = []
 
+    msg_to_str = ' '.join(str(uuid) for uuid in msg_to)
```
Comment (rjweeks) [inline app/common/utilities.py]: Is there a unit test for multiple uuid's in msg_to?

File: app/validation/domain.py [hunk 2]
```diff
@@ -41,7 +41,8 @@ class MessageSchema(Schema):
 
     """ Class to marshal JSON to Message"""
     msg_id = fields.Str(allow_none=True)
-    msg_to = fields.Str(required=True)
+    # msg_to = fields.Str(required=True)
```
Comment (rjweeks) [inline app/validation/domain.py]: Remove commented out code.

File: app/validation/domain.py [hunk 3]
```diff
@@ -57,34 +58,36 @@ def check_sent_and_read_date(self, data):
...
+    #             data['msg_to'] = data.get('msg_to')[0]
+    #         else:
+    #             raise ValidationError("'msg_to' is missing an 'id' or incorrect format")
+    #
+    #     if data.get('msg_from') and not isinstance(data.get('msg_from'), str) and "id" in data.get('msg_from'):
+    #         data['msg_from'] = data['msg_from']['id']
+    #     elif data.get('msg_from') and not isinstance(data.get('msg_from'), str):
+    #         raise ValidationError("'msg_from' is missing an 'id' or incorrect format")
+    #
+    #     return data
```
Comment (rjweeks) [inline app/validation/domain.py]: Remove commented out code.

File: app/validation/domain.py [hunk 4]
```diff
@@ -57,34 +58,36 @@ def check_sent_and_read_date(self, data):
...
 
     @validates_schema
     def validate_to_from_not_equal(self, data):
-        if 'msg_to' in data.keys() and 'msg_from' in data.keys() and data['msg_to'] == data['msg_from']:
+        if 'msg_to' in data.keys() and 'msg_from' in data.keys() and data['msg_to'][0] == data['msg_from']:
             raise ValidationError("msg_to and msg_from fields can not be the same.")
 
+    # TODO: Validate UUID wit mock party
```
Comment (rjweeks) [inline app/validation/domain.py]: typo: 'wit' should be 'with'

File: app/validation/domain.py [hunk 5]
```diff
@@ -152,7 +155,8 @@ class DraftSchema(Schema):
     """Class to marshal JSON to Draft"""
 
     msg_id = fields.Str(allow_none=True)
-    msg_to = fields.Str(allow_none=True)
+    # msg_to = fields.Str(allow_none=True)
```
Comment (rjweeks) [inline app/validation/domain.py]: Remove commented out code.

File: app/validation/domain.py [hunk 6]
```diff
@@ -171,26 +175,26 @@ def check_variables_set_and_not_set(self, data):
...
+    #             raise ValidationError("'msg_to' is missing an 'id' or incorrect format")
+    #     elif 'msg_to' in data.keys() and isinstance(data.get('msg_to'), list) and len(data.get('msg_to')) == 0:
+    #         data.pop('msg_to')
+    #
+    #     if data.get('msg_from') and not isinstance(data.get('msg_from'), str) and "id" in data.get('msg_from'):
+    #         data['msg_from'] = data['msg_from']['id']
+    #     elif data.get('msg_from') and not isinstance(data.get('msg_from'), str):
+    #         raise ValidationError("'msg_from' is missing an 'id' or incorrect format")
+    #
+    #     return data
```
Comment (rjweeks) [inline app/validation/domain.py]: Remove commented out code.

File: app/validation/domain.py [hunk 7]
```diff
@@ -199,10 +203,14 @@ def validate_to_from_not_equal(self, data):
 
     @validates("msg_to")
     def validate_to(self, msg_to):
-        if msg_to is not None:
-            self.validate_field_length(msg_to, len(msg_to), constants.MAX_TO_LEN)
-            if len(msg_to) > 0 and msg_to != 'BRES' and not User.is_valid_user(msg_to):
-                raise ValidationError("{0} is not a valid user.".format(msg_to))
+        if msg_to:
+        # if msg_to is not None:
```
Comment (rjweeks) [inline app/validation/domain.py]: Remove commented out code.

File: app/validation/domain.py [hunk 8]
```diff
@@ -199,10 +203,14 @@ def validate_to_from_not_equal(self, data):
...
+            for item in msg_to:
+                self.validate_field_length(msg_to, len(item), constants.MAX_TO_LEN)
+                if len(msg_to) > 0 and msg_to != 'BRES' and not User.is_valid_user(item):
+                    raise ValidationError("{0} is not a valid user.".format(item))
+        else:
+            logger.debug("")
```
Comment (rjweeks) [inline app/validation/domain.py]: Sensible debug statement here.

**Comment by emilioward** [comment]
[TS-204 Modify User_to to enforce that the to is in the form of a list . Do not support a string as the to param |13|](https://trello.com/c/iZMW18em/368-ts-204-modify-userto-to-enforce-that-the-to-is-in-the-form-of-a-list-do-not-support-a-string-as-the-to-param-13)

**Pull request merged**